modules/pam_unix/support.c

```diff
@@ -515,6 +518,89 @@ int _unix_comesfromsource(pam_handle_t *pamh,
 #include <sys/types.h>
 #include <sys/wait.h>
 
+#ifdef USE_PWACCESS
+int _unix_verify_pwaccess(pam_handle_t *pamh, const char *user,
+			  int *daysleft)
+{
+    int  r, retval = PAM_SYSTEM_ERR;
+    bool pwchangeable = false;
+    long dl = -1;
+
+    D(("called."));
+
+    r = pwaccess_check_expired(user, &dl, &pwchangeable, NULL);
+    if (r < 0) {
+	if (PWACCESS_IS_NOT_RUNNING(r))
+	    return PAM_SYSTEM_ERR;
+
+	if (r == -ENOENT)
+	    return PAM_USER_UNKNOWN;
+
+	pam_syslog(pamh, LOG_ERR, "pwaccess failed: %s", strerror(-r));
+	return PAM_AUTHINFO_UNAVAIL;
+    }
+
+    if (dl > INT_MAX)
+        *daysleft = INT_MAX;
+    else
+        *daysleft = dl;
+
+    switch(r) {
+        case PWA_EXPIRED_NO:
+	    if (!pwchangeable) {
+	        D(("password change too recent"));
+                return PAM_AUTHTOK_ERR;
+	    }
+	    retval = PAM_SUCCESS;
+	    break;
+        case PWA_EXPIRED_YES:
+            retval = PAM_ACCT_EXPIRED;
+            break;
+        case PWA_EXPIRED_CHANGE_PW:
+            retval = PAM_NEW_AUTHTOK_REQD;
+            break;
+        case PWA_EXPIRED_DISABLED:
+            retval = PAM_AUTHTOK_EXPIRED;
+            break;
+        default:
+	    retval = PAM_SYSTEM_ERR;
+    }
+
+    return retval;
+}
+
+static int _unix_use_pwaccess(pam_handle_t *pamh, const char *passwd,
+			      unsigned long long ctrl, const char *user)
+{
+    int  r;
+    bool authenticated = false;
+    bool nullok = false;
+
+    D(("called."));
+
+    if (off(UNIX__NONULL, ctrl))
+        nullok = true;
+
+
+    r = pwaccess_verify_password(user, passwd, nullok, &authenticated, NULL);
```

Comment:
Looks like `pwaccess_verify_password` can return a textual error. Is there any reason why this feature is not used here?